lib/http/request.js

```diff
@@ -9,6 +10,11 @@ const HttpResponse = require('./response.js');
 const Utils = require('./../utils');
 const {Logger, isString} = Utils;
 
+// To handle Node v17 issue. Refer https://github.com/nodejs/node/issues/40702 for details.
+if (dns.setDefaultResultOrder && (typeof dns.setDefaultResultOrder === 'function')) {
+  dns.setDefaultResultOrder('ipv4first');
```

Comment:
https://github.com/nodejs/node/issues/40702#issuecomment-967270434 seems to make a fair point about this fix not accounting for ipv6 in the future. Can you please revisit and see how to tackle that bit? Thank you!

Reply:
Thank you for pointing this out. We can't hardcode `localhost` to `127.0.0.1` as we have added nocks for localhost and that change would result in changing a lot of tests. Node.js now returns IP addresses in the order they are returned from the name resolver/DNS.  This was changed from Node version 17. This command sets back the default order. It will still account for the ipv6 address. For more details refer to this [issue](https://github.com/nodejs/node/issues/40537)

Reply:
Thank you for checking and providing the context. As long as it accounts for ipv6 address, the PR looks good to me.